apps/analysis/tests/test_mutations.py

```diff
@@ -1326,3 +1326,204 @@ def _query_check(_id, **kwargs):
+                id
+                endDate
+                title
+                teamLead {
+                  id
+                }
+              }
+            }
+          }
+        }
+    '''
+    DELETE_MUTATION = '''
+        mutation MyMutation($projectId: ID!, $deleteId: ID!) {
+          project(id: $projectId) {
+            analysisDelete(id: $deleteId) {
+              errors
+              result {
+                id
+                title
+              }
+            }
+          }
+        }
+    '''
+
+    def setUp(self):
+        super().setUp()
+        self.af = AnalysisFrameworkFactory.create()
+        self.project_with_af = ProjectFactory.create(analysis_framework=self.af)
+        self.project_without_af = ProjectFactory.create()
+        # Users with different roles
+        self.non_member_user = UserFactory.create()
+        self.readonly_member_user = UserFactory.create()
+        self.member_user = UserFactory.create()
+        self.project_with_af.add_member(self.readonly_member_user, role=self.project_role_reader_non_confidential)
+        self.project_with_af.add_member(self.member_user, role=self.project_role_member)
+
+    def test_create_analysis_without_pillar(self):
+        minput = dict(
+            analysisData=dict(
+                title='Test Analysis', teamLead=self.member_user.id, endDate='2020-01-01'
+            ),
+            projectId=self.project_with_af.id,
+        )
+        # -- Without login
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
+
+        # -- With login (non-member)
+        self.force_login(self.non_member_user)
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
+
+        # --- member user (read-only)
+        self.force_login(self.readonly_member_user)
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
+
+        # --- member user
+        self.force_login(self.member_user)
+        content = self.query_check(self.CREATE_MUTATION, variables=minput)
+
+        self.assertEqual(
+            content['data']['project']['analysisCreate']['result']['title'],
+            minput['analysisData']['title']
+        )
+        self.assertEqual(
+            content['data']['project']['analysisCreate']['result']['teamLead']['id'],
+            str(self.member_user.id)
+        )
+        self.assertEqual(
+            content['data']['project']['analysisCreate']['result']['endDate'],
+            str(minput['analysisData']['endDate'])
+        )
+
+    def test_create_analysis_without_analysis_framework(self):
+        minput = dict(
+            analysisData=dict(
+                title='Test Analysis', teamLead=self.member_user.id, endDate='2020-01-01'
+            ),
+            projectId=self.project_without_af.id,
+        )
+
+        self.force_login(self.member_user)
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
```

Comment:
We can add this simple behaviour checking inside the `test_create_and_update_analysis` Function. 
Let's use DRY method.
> Add comment before checking.